source/utils/Helper.h

```diff
@@ -30,6 +30,10 @@
 #include <vector>
 #include <cstdint>
 
+#ifdef _MSC_VER
+    #define snprintf _snprintf
```

Comment:
They're functionally the same, but snprintf allows you to do it on the stack instead of on the heap. However, if the "No C" rule prevails, I'm perfectly willing to use the stringstream version instead.